src/main/java/org/spongepowered/api/event/resource/ResourceEvent.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.resource;
+
+import org.spongepowered.api.event.Event;
+import org.spongepowered.api.resource.ReloadableResourceManager;
+import org.spongepowered.api.resource.ResourceReloadListener;
+import org.spongepowered.api.resource.Resource;
+import org.spongepowered.api.resource.ResourceManager;
+
+/**
+ * Base event for {@link Resource}s.
+ */
+public interface ResourceEvent extends Event {
+
+    /**
+     * Gets the relevant {@link ResourceManager} for this event.
+     *
+     * @return The resource manager
+     */
+    ResourceManager getResourceManager();
+
+    /**
+     * Event for registering {@link ResourceReloadListener reload listeners} to the
+     * {@link ResourceManager}. It is fired sometime during init.
+     */
+    interface RegisterReloadListener extends ResourceEvent {
+
+        /**
+         * {@inheritDoc}
+         *
+         * <p>The type is a {@link ReloadableResourceManager} to expose methods
+         * to add reload listeners</p>
+         *
+         * @return The reloadable resource manager
+         */
+        @Override
+        ReloadableResourceManager getResourceManager();
+
+        /**
+         * Register event for client resources.
+         */
+        interface Client extends RegisterReloadListener {
```

Comment:
Same as the above, a generic engine event.

If lifecycle, put it there along with the above.

Reply:
I wouldn't consider it a lifecycle event. It's just a specific event that does a specific task once at a specific time during init.